components/eam/src/physics/cam/aer_rad_props.F90

```diff
@@ -262,8 +265,13 @@ subroutine aer_rad_props_sw(list_idx, state, pbuf,  nnite, idxnite, is_cmip6_vol
 
    ! Contributions from modal aerosols.
    if (nmodes > 0) then
-      call modal_aero_sw(list_idx, state, pbuf, nnite, idxnite, is_cmip6_volc, ext_cmip6_sw(:,:,idx_sw_diag), trop_level, &
-                         tau, tau_w, tau_w_g, tau_w_f)
+      if(present(clear_rh)) then
```

Comment:
I think this extra logic is unnecessary; you can just pass clear_rh as an optional and `model_aero_sw` will determine if it is a valid argument.